added session on GitHub READMES (#79)
NBISweden/Training-Tech-shorts: posts/2025-03-18-github-readmes/index.qmd

## profile README

Every personal account can have its own README, which is displayed on top of the profile page. This is a place to share information about you and the work you are doing. 

To initialize the profile README 

- create a repository with a name that matches your GitHub user name
- make it public
- add a README

More information on the GitHub documentation pages [GitHub documentation pages](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/managing-your-profile-readme).

# Dressing up the README

Regardless of which type of README, there are some things that can be used to make the page more informative and break up the text. Most of the below options were developed for profile README's, but they can also be used on all kinds of repositories.

## 😎 Add emojis

You can use emojis in the readme to clarify points (and even add them to commits, as I have just learned, use cases are to be determined). 

- 🕰 You can point out deadlines

> - [📕](https://www.webfx.com/tools/emoji-cheat-sheet/) Link to documentation

Reviewer: Perhaps include the text in the link too.

Author: not sure what you mane here - it's the same link that is also linked below.

Reviewer: ```suggestion
- [📕 Link to documentation](https://www.webfx.com/tools/emoji-cheat-sheet/)
```
Is what I meant.

> - ✏ Or invite them to contribute

Reviewer: I can't tell what this emoji is. Initially I thought it was a spill on my screen.

```html
<a href="https://github.com/NBISweden/Training-Tech-shorts/graphs/contributors">
  <img src="https://contrib.rocks/image?repo=NBISweden/Training-Tech-shorts" />
</a>
```

<a href="https://github.com/NBISweden/Training-Tech-shorts/graphs/contributors">
  <img src="https://contrib.rocks/image?repo=NBISweden/Training-Tech-shorts" />
</a>

---

## License

Make it easy for others to know what they can do with your code and add license information into your readme. Often these are placed at the top of the readme for even easier accessibility. They link to the text of the license (not in your repository, but on the web). 

[![](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)

Here are some [markdown license badges](https://gist.github.com/lukas-h/2a5d00690736b4c3a7ba) you can use.

---

## Add images

You can add images to break up the text in your README. 

### Title banner

These you can use for projects, but also for your very own user readme. They are easily generated [here](https://leviarista.github.io/github-profile-header-generator/). 

![](/posts/2025-03-18-github-readmes/github-header-image.png)

### Octocat

> You can [design your very own octocat](https://myoctocat.com/), or [choose an existing one](https://octodex.github.com/),  and display it in your readme: 

Reviewer: Can you add code examples of what the markdown should be please to add these things?

An example of a static badge:

![](https://img.shields.io/badge/this_is_a_test-badge-blue)

An example of a dynamic badge: 

![](https://img.shields.io/github/directory-file-count/amrei-bp/readme_playground?label=Number%20of%20files%20in%20repository%3A)

And another one: 

![](https://img.shields.io/github/last-commit/amrei-bp/readme_playground?label=Date%20of%20last%20commit%20to%20the%20repository%3A%20)

And another one:

This counts the number of times a Github profile has been accessed since the tracker was installed:

[![](https://komarev.com/ghpvc/?username=amrei-bp&color=blue&label=GitHub+Profile+Views+Amrei)](https://github.com/amrei-bp)

[![](https://komarev.com/ghpvc/?username=NBISweden&color=blue&label=GitHub+Profile+Views+NBIS)](https://github.com/NBISweden)

---

## Animated text

You can have text appear on your page, as if it was typed on the spot:

[![](https://readme-typing-svg.demolab.com/?lines=First+line+of+text;Second+line+of+text)](https://git.io/typing-svg)

Customizable [here](https://readme-typing-svg.demolab.com/demo/).

---

## Link to the readme_playground repository

> Here is my [README playground repository](https://github.com/amrei-bp/readme_playground) which contains a few more things that do not render easily in quarto. 

Author: Basically only the snake game and another set of skill icons. I removed it now, because I don't think it's ultimately very useful.

---
title: "README's in GitHub"
author: "Amrei"
date: "2025-03-18"
date-modified: last-modified

> categories: [Github, documentation, self-promotion]

Reviewer: ```suggestion
categories: [Github, Documentation, Self-promotion]
```

<p align="center">
  <a href="https://go-skill-icons.vercel.app/">
    <img src="https://go-skill-icons.vercel.app/api/icons?i=git,github,apptainer,docker,nextflow,cpp,R&theme=light&titles=true" />
  </a>
</p>

Below is an example on how to display skill icons in dark and light mode, depending on the system settings: 

```html
<div align="center">
  
  <!-- Dark Mode -->
  [![My skills](https://go-skill-icons.vercel.app/api/icons?i=linux,git,apptainer,anaconda,nextflow,md,vscode,latex,r,bash,cpp&theme=dark#gh-dark-mode-only)](https://github.com/LelouchFR/skill-icons#gh-dark-mode-only)
  <!-- Light Mode -->
  [![My skills](https://go-skill-icons.vercel.app/api/icons?i=linux,git,apptainer,anaconda,nextflow,md,vscode,latex,r,bash,cpp&theme=light#gh-light-mode-only)](https://github.com/LelouchFR/skill-icons#gh-light-mode-only)

</div>
```

---

## Add Github statistics

There are several different ways to display your github stats (if you want to). By default, they will take information off your public repositories, but it is possible to gather information from the private ones too. They come pre-defined for several statistics, but some are customizable.

> The [Github Profile Summary Cards](https://github-profile-summary-cards.vercel.app/demo.html), for example, come in a lot of different [themes](https://github.com/vn7n24fzkq/github-profile-summary-cards/blob/main/README.md) and are easily incorporated. Let's have a look at Mahesh's statistics in slateorange:

Reviewer: ```suggestion
The [Github Profile Summary Cards](https://github-profile-summary-cards.vercel.app/demo.html), for example, come in a lot of different [themes](https://github.com/vn7n24fzkq/github-profile-summary-cards/blob/main/README.md) and are easily incorporated. Let's have a look at Mahesh's statistics in slate-orange:
```
